### PR TITLE
Adds actual build status from docker hub as discussed in issue #241

### PR DESCRIPTION
**Pull request by eddiewebb**
Building on @jrottenberg 's contribution, but adding actual status for latest build to address discussion in issue #241 

screen shot 2017-01-14 at 8 56 55 am

I debated just leaving the language to duplicate "Docker Automated Build" and let's folks choose. BUt ultimately decided to use "Docker Build Status" instead, I'd use it either way.

### CODE REVIEW

**Comment by jrottenberg** [comment]
Oh ! Very nice

**Comment by eddiewebb** [comment]
@espadrine  - anything I can do to help address the backlog of pull requests?

**Comment by Bekt** [comment]
Anything holding this?

File: server.js [hunk 1]
```diff
@@ -4960,6 +4960,45 @@ cache(function(data, match, sendBadge, request) {
...
+      var data = JSON.parse(buffer);
+      var latest_status = data.results[0].status;
+      if ( latest_status == 10 ) {
+        badgeData.text[1] = 'passing';
+        badgeData.colorscheme = 'brightgreen';
+      } else if ( latest_status < 0 ) {
+        badgeData.text[1] = 'failing';
+        badgeData.colorscheme = 'red';
+      } else {
+        badgeData.text[1] = 'building..';
```
Comment (paulmelnikow) [inline server.js]: Could you change this to `'building'`, for consistency with the appearance of other badges at https://shields.io/?

File: server.js [hunk 2]
```diff
@@ -4960,6 +4960,45 @@ cache(function(data, match, sendBadge, request) {
...
+  request(url, function(err, res, buffer) {
+    if (err != null) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+      return;
+    }
+    try {
+      var data = JSON.parse(buffer);
+      var latest_status = data.results[0].status;
+      if ( latest_status == 10 ) {
```
Comment (paulmelnikow) [inline server.js]: Can you remove the spacing in this `if`? Most of the code looks that way today, though we don't have that linter rule turned on just yet.

File: server.js [hunk 3]
```diff
@@ -4960,6 +4960,45 @@ cache(function(data, match, sendBadge, request) {
...
+        badgeData.text[1] = 'passing';
+        badgeData.colorscheme = 'brightgreen';
+      } else if ( latest_status < 0 ) {
+        badgeData.text[1] = 'failing';
+        badgeData.colorscheme = 'red';
+      } else {
+        badgeData.text[1] = 'building..';
+        badgeData.colorB = '#008bb8';
```
Comment (paulmelnikow) [inline server.js]: I like that color!

File: server.js [hunk 4]
```diff
@@ -4960,6 +4960,45 @@ cache(function(data, match, sendBadge, request) {
...
+cache(function(data, match, sendBadge, request) {
+  var user = match[1];  // eg, jrottenberg
+  var repo = match[2];  // eg, ffmpeg
+  var format = match[3];
+  if (user === '_') {
+    user = 'library';
+  }
+  var path = user + '/' + repo;
+  var url = 'https://registry.hub.docker.com/v2/repositories/' + path + '/buildhistory';
+  var badgeData = getBadgeData('docker status', data);
```
Comment (paulmelnikow) [inline server.js]: `'docker build'` seems like a clearer label, which would be more consistent with the other CI badges.

File: server.js [hunk 5]
```diff
@@ -4960,6 +4960,45 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// Docker Hub automated integration, latest status (passed, pending, failed)
+camp.route(/^\/docker\/status\/([^\/]+)\/([^\/]+)\.(svg|png|gif|jpg|json)$/,
```
Comment (paulmelnikow) [inline server.js]: How about using `docker/build` for this instead of `docker/status`?

**Comment by paulmelnikow** [comment]
Hi, thanks for this contribution! Would love to get this merged. Left a handful of minor comments. If you could also merge master into the PR branch, you can check that the automated tests are passing.

**Comment by paulmelnikow** [comment]
@eddiewebb I just saw your note above! Would be great to have some more help with the project.  :smile:

Would you be interested in working on some automated tests?

**Comment by eddiewebb** [comment]
I thought docker/build conflicts with the existing shield (which just indicates whether it exists, not health). But I can confirm and make the other changes.

If it does conflict, would you rather replace or use a different path?

**Comment by paulmelnikow** [comment]
The current badge is at `/docker/automated`. You can type in "docker" at  the top of https://shields.io/ to filter the listing, which is up to date with what's in `server.js`.

**Comment by eddiewebb** [comment]
Beauty. You'll see updates shortly

**Comment by eddiewebb** [comment]
Sorry, didn't get to it last night, but made the changes you suggested and merged in latest from the upstream's master branch.

**Comment by eddiewebb** [comment]
screen shot 2017-04-06 at 9 40 22 am

screen shot 2017-04-06 at 9 40 07 am

**Comment by paulmelnikow** [comment]
Those error badges look a little busy, how about just `'repo not found'`?

This is looking great. I'll test on my machine and then should be good to go.

**Comment by eddiewebb** [comment]
new look for your review:
screen shot 2017-04-06 at 11 14 14 am

**Comment by paulmelnikow** [comment]
Love it! All looks good. Last step is to test this myself, which might take a day or two.

**Comment by paulmelnikow** [comment]
Cool, ran this and it looks good.

This is a documentation nit. From looking at https://hub.docker.com/r/jrottenberg/ffmpeg/builds/, it seems like "latest" has a special meaning on Docker Hub. Is that correct? If so, would it be better to avoid the word "latest" in the code and documentation here, and instead say something like "most recent"?

**Comment by eddiewebb** [comment]
@paulmelnikow, it is a convention many use to Tag the most recent stable build as "LATEST".

I'll update the code and comments to be most_recent to eliminate any ambiguity.

**Comment by paulmelnikow** [comment]
Thanks! Appreciate your responsiveness.

**Comment by niccokunzmann** [comment]
- [ ] Should index.html also be changed andcommitted? `make website` should be the command to do that.
- [ ] When will the changes go live?

**Comment by paulmelnikow** [comment]
index.html: I'd prefer we just update and commit it at deploy time. It makes PRs easier to read, avoids unnecessary merge conflicts, is fewer steps for contributors, and makes it less likely contributors will accidentally update `index.html` instead of `try.html`.

I'd like to update the deploy script to do that, feel free to open an issue if you'd like to track it.

The changes will go live when Thaddée next deploys the server. How's that for a tautology?

**Comment by niccokunzmann** [comment]
@paulmelnikow I created two issues for discussion and checked all questions as answered. Thanks!

**Comment by hobakill** [comment]
@paulmelnikow @eddiewebb really been looking forward to this. thanks! even though it's been merged in is there a reason it's not showing up on the homepage? 

forgive me if i'm missing something obvious.

**Comment by niccokunzmann** [comment]
@hobakill 

> The changes will go live when Thaddée next deploys the server. 

[source](https://github.com/badges/shields/pull/856#issuecomment-292545740)

**Comment by paulmelnikow** [comment]
I'm glad everyone is so excited about it! I'll post a note here once it happens.

**Comment by espadrine** [comment]
:rocket: Deployed.

**Comment by sinedied** [comment]
=> works fine
 => not found

Am I missing something?

**Comment by paulmelnikow** [comment]
Could you open a new issue, please?

**Comment by sinedied** [comment]
Sure, will do.

**Comment by liaogz82** [comment]
what is the markdown code to implement these docker badge?

**Comment by RedSparr0w** [comment]
@liaogz82:
`Docker Badge`
Replace `<type>`, `<user>`, `<repo>` with your own values, eg:
```markdown
![Docker Stars](https://img.shields.io/docker/stars/jrottenberg/ffmpeg.svg)
![Docker Pulls](https://img.shields.io/docker/pulls/jrottenberg/ffmpeg.svg)
![Docker Automated](https://img.shields.io/docker/automated/jrottenberg/ffmpeg.svg)
![Docker Build](https://img.shields.io/docker/build/jrottenberg/ffmpeg.svg)
```
Docker Stars
Docker Pulls
Docker Automated
Docker Build

**Comment by liaogz82** [comment]
Not working. It says error fetching resources

**Comment by eddiewebb** [comment]
Please open an issue.

**Comment by thiagospassos** [comment]
For me it show `manual` even though it's automated

**Comment by paulmelnikow** [comment]
Hi! This PR is two years old. Could you please open a new issue?

**Pull request merged**